From: Hannes Reinecke <hare@suse.de>
Date: Thu, 2 Feb 2012 15:19:44 +0100
Subject: qla4xxx: Do not print warning when rom lock fails
References: bnc#744652
Patch-Mainline: not yet

When the driver can not acquire the hardware lock it
shouldn't print out a kernel message as normally
ql4_8xxx_rom_lock() will be called in a loop, with an
additional warning if the lock couldn't be acquired
within a given timeout. So printing a message here
will just explode the system without any additional
benefit.

Signed-off-by: Hannes Reinecke <hare@suse.de>

diff --git a/drivers/scsi/qla4xxx/ql4_nx.c b/drivers/scsi/qla4xxx/ql4_nx.c
index 78f1111..9917ea9 100644
--- a/drivers/scsi/qla4xxx/ql4_nx.c
+++ b/drivers/scsi/qla4xxx/ql4_nx.c
@@ -860,11 +860,8 @@ qla4_8xxx_rom_lock(struct scsi_qla_host *ha)
 		done = qla4_8xxx_rd_32(ha, QLA82XX_PCIE_REG(PCIE_SEM2_LOCK));
 		if (done == 1)
 			break;
-		if (timeout >= qla4_8xxx_rom_lock_timeout) {
-			ql4_printk(KERN_WARNING, ha,
-			    "%s: Failed to acquire rom lock", __func__);
+		if (timeout >= qla4_8xxx_rom_lock_timeout)
 			return -1;
-		}
 
 		timeout++;
 
